nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/events/BulletinFactory.java

```diff
@@ -126,4 +177,17 @@ private static ComponentType getComponentType(final Connectable connectable) {
             default -> ComponentType.PROCESSOR;
         };
     }
+
+    private static String formatStackTrace(final Throwable t) {
+        try {
+            final java.io.StringWriter sw = new java.io.StringWriter();
+            final java.io.PrintWriter pw = new java.io.PrintWriter(sw);
```

Comment:
Is there a reason for using the fully-qualified class names? Although it should not matter with this in-memory usage, should the Writers be declared in a try-with-resources block?